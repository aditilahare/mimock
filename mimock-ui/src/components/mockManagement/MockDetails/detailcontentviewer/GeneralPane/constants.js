export const constants = {
	testIds: {
		generalNavTabPane: 'general-nav-tab-pane',
		mockId: 'mock-id',
		mockName: 'mock-name',
		description: 'mock-description',
		entityStatus: 'mock-entity-status',
		httpMethod: 'mock-http-method',
		route: 'mock-route',
		queryParams: 'mock-query-params',
		statusCode: 'mock-status-code',
		createdBy: 'mock-created-by',
		createdAt: 'mock-created-at',
		lastModifiedBy: 'mock-last-modified-by',
		lastModifiedAt: 'mock-last-modified-at',
	},
	labels: {
		mockId: 'Mock Unique Identifier',
		mockName: 'Mock Name',
		description: 'Description',
		entityStatus: 'Status',
		httpMethod: 'HTTP Method',
		route: 'Route',
		queryParams: 'Query Params',
		statusCode: 'Expected Status Code',
		createdBy: 'Created By',
		createdAt: 'Created At',
		lastModifiedBy: 'Last Modified By',
		lastModifiedAt: 'Last Modified At',
	},
	defaultFieldType: 'text',
};
