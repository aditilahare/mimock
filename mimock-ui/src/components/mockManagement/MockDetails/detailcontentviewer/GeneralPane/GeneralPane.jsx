import React from 'react';
import { NavTabPane, ContentItem, ItemLabel, Item } from './GeneralPane.style';
import PropTypes from 'prop-types';
import { constants } from './constants';

function GeneralPane({
	uniqueId,
	mockName,
	description,
	entityStatus,
	httpMethod,
	route,
	queryParams,
	statusCode,
	createdBy,
	createdAt,
	modifiedBy,
	updatedAt,
}) {
	return (
		<NavTabPane data-testid={constants.testIds.generalNavTabPane}>
			<ContentItem>
				<ItemLabel>{constants.labels.mockId}</ItemLabel>
				<Item
					data-testid={constants.testIds.mockId}
					type={constants.defaultFieldType}
					defaultValue={uniqueId}
					disabled
					readOnly
				/>
			</ContentItem>
			<ContentItem>
				<ItemLabel>{constants.labels.mockName}</ItemLabel>
				<Item
					data-testid={constants.testIds.mockName}
					type={constants.defaultFieldType}
					defaultValue={mockName}
					disabled
					readOnly
				/>
			</ContentItem>
			<ContentItem>
				<ItemLabel>{constants.labels.description}</ItemLabel>
				<Item
					data-testid={constants.testIds.description}
					type={constants.defaultFieldType}
					defaultValue={description}
					disabled
					readOnly
				/>
			</ContentItem>
			<ContentItem>
				<ItemLabel>{constants.labels.entityStatus}</ItemLabel>
				<Item
					data-testid={constants.testIds.entityStatus}
					type={constants.defaultFieldType}
					defaultValue={entityStatus}
					disabled
					readOnly
				/>
			</ContentItem>
			<ContentItem>
				<ItemLabel>{constants.labels.httpMethod}</ItemLabel>
				<Item
					data-testid={constants.testIds.httpMethod}
					type={constants.defaultFieldType}
					defaultValue={httpMethod}
					disabled
					readOnly
				/>
			</ContentItem>
			<ContentItem>
				<ItemLabel>{constants.labels.route}</ItemLabel>
				<Item
					data-testid={constants.testIds.route}
					type={constants.defaultFieldType}
					defaultValue={route}
					disabled
					readOnly
				/>
			</ContentItem>
			<ContentItem>
				<ItemLabel>{constants.labels.queryParams}</ItemLabel>
				<Item
					data-testid={constants.testIds.queryParams}
					type={constants.defaultFieldType}
					defaultValue={queryParams}
					disabled
					readOnly
				/>
			</ContentItem>
			<ContentItem>
				<ItemLabel>{constants.labels.statusCode}</ItemLabel>
				<Item
					data-testid={constants.testIds.statusCode}
					type={constants.defaultFieldType}
					defaultValue={statusCode}
					disabled
					readOnly
				/>
			</ContentItem>
			<ContentItem>
				<ItemLabel>{constants.labels.createdBy}</ItemLabel>
				<Item
					data-testid={constants.testIds.createdBy}
					type={constants.defaultFieldType}
					defaultValue={createdBy}
					disabled
					readOnly
				/>
			</ContentItem>
			<ContentItem>
				<ItemLabel>{constants.labels.createdAt}</ItemLabel>
				<Item
					data-testid={constants.testIds.createdAt}
					type={constants.defaultFieldType}
					defaultValue={createdAt}
					disabled
					readOnly
				/>
			</ContentItem>
			<ContentItem>
				<ItemLabel>{constants.labels.lastModifiedBy}</ItemLabel>
				<Item
					data-testid={constants.testIds.lastModifiedBy}
					type={constants.defaultFieldType}
					defaultValue={modifiedBy}
					disabled
					readOnly
				/>
			</ContentItem>
			<ContentItem>
				<ItemLabel>{constants.labels.lastModifiedAt}</ItemLabel>
				<Item
					data-testid={constants.testIds.lastModifiedAt}
					type={constants.defaultFieldType}
					defaultValue={updatedAt}
					disabled
					readOnly
				/>
			</ContentItem>
		</NavTabPane>
	);
}

GeneralPane.propTypes = {
	uniqueId: PropTypes.string.isRequired,
	mockName: PropTypes.string.isRequired,
	description: PropTypes.string.isRequired,
	entityStatus: PropTypes.string.isRequired,
	httpMethod: PropTypes.string.isRequired,
	route: PropTypes.string.isRequired,
	queryParams: PropTypes.string.isRequired,
	statusCode: PropTypes.number.isRequired,
	createdBy: PropTypes.string.isRequired,
	createdAt: PropTypes.string.isRequired,
	modifiedBy: PropTypes.string.isRequired,
	updatedAt: PropTypes.string.isRequired,
};

export default GeneralPane;
