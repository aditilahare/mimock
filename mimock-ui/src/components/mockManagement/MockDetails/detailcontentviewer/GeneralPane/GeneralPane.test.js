import React from 'react';
import { render } from '@testing-library/react';
import GeneralPane from './GeneralPane';
import { BrowserRouter } from 'react-router-dom';

describe('GeneralPane', () => {
	it('should render general pane component', async () => {
		const tree = await render(<GeneralPane />, { wrapper: BrowserRouter });

		const { container, getByTestId } = tree;

		expect(getByTestId('general-nav-tab-pane')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});
