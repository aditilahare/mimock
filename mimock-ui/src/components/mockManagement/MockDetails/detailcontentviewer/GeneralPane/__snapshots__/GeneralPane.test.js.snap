// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralPane should render general pane component 1`] = `
<div>
  <div
    class="grid grid-cols-2 gap-6 fade px-4 py-2 my-2"
    data-testid="general-nav-tab-pane"
  >
    <div
      class="mb-3"
    >
      <label
        class="form-label inline-block mb-2 text-gray-700"
      >
        Mock Unique Identifier
      </label>
      <input
        class="form-control block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-gray-100 bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:outline-none"
        data-testid="mock-id"
        disabled=""
        readonly=""
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label inline-block mb-2 text-gray-700"
      >
        Mock Name
      </label>
      <input
        class="form-control block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-gray-100 bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:outline-none"
        data-testid="mock-name"
        disabled=""
        readonly=""
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label inline-block mb-2 text-gray-700"
      >
        Description
      </label>
      <input
        class="form-control block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-gray-100 bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:outline-none"
        data-testid="mock-description"
        disabled=""
        readonly=""
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label inline-block mb-2 text-gray-700"
      >
        Status
      </label>
      <input
        class="form-control block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-gray-100 bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:outline-none"
        data-testid="mock-entity-status"
        disabled=""
        readonly=""
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label inline-block mb-2 text-gray-700"
      >
        HTTP Method
      </label>
      <input
        class="form-control block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-gray-100 bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:outline-none"
        data-testid="mock-http-method"
        disabled=""
        readonly=""
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label inline-block mb-2 text-gray-700"
      >
        Route
      </label>
      <input
        class="form-control block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-gray-100 bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:outline-none"
        data-testid="mock-route"
        disabled=""
        readonly=""
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label inline-block mb-2 text-gray-700"
      >
        Query Params
      </label>
      <input
        class="form-control block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-gray-100 bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:outline-none"
        data-testid="mock-query-params"
        disabled=""
        readonly=""
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label inline-block mb-2 text-gray-700"
      >
        Expected Status Code
      </label>
      <input
        class="form-control block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-gray-100 bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:outline-none"
        data-testid="mock-status-code"
        disabled=""
        readonly=""
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label inline-block mb-2 text-gray-700"
      >
        Created By
      </label>
      <input
        class="form-control block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-gray-100 bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:outline-none"
        data-testid="mock-created-by"
        disabled=""
        readonly=""
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label inline-block mb-2 text-gray-700"
      >
        Created At
      </label>
      <input
        class="form-control block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-gray-100 bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:outline-none"
        data-testid="mock-created-at"
        disabled=""
        readonly=""
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label inline-block mb-2 text-gray-700"
      >
        Last Modified By
      </label>
      <input
        class="form-control block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-gray-100 bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:outline-none"
        data-testid="mock-last-modified-by"
        disabled=""
        readonly=""
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label inline-block mb-2 text-gray-700"
      >
        Last Modified At
      </label>
      <input
        class="form-control block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-gray-100 bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:outline-none"
        data-testid="mock-last-modified-at"
        disabled=""
        readonly=""
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
